stl/inc/bitset

```diff
@@ -336,29 +337,36 @@ public:
     }
 
     _NODISCARD size_t count() const noexcept { // count number of set bits
-        const char* const _Bitsperbyte = "\0\1\1\2\1\2\2\3\1\2\2\3\2\3\3\4"
-                                         "\1\2\2\3\2\3\3\4\2\3\3\4\3\4\4\5"
-                                         "\1\2\2\3\2\3\3\4\2\3\3\4\3\4\4\5"
-                                         "\2\3\3\4\3\4\4\5\3\4\4\5\4\5\5\6"
-                                         "\1\2\2\3\2\3\3\4\2\3\3\4\3\4\4\5"
-                                         "\2\3\3\4\3\4\4\5\3\4\4\5\4\5\5\6"
-                                         "\2\3\3\4\3\4\4\5\3\4\4\5\4\5\5\6"
-                                         "\3\4\4\5\4\5\5\6\4\5\5\6\5\6\6\7"
-                                         "\1\2\2\3\2\3\3\4\2\3\3\4\3\4\4\5"
-                                         "\2\3\3\4\3\4\4\5\3\4\4\5\4\5\5\6"
-                                         "\2\3\3\4\3\4\4\5\3\4\4\5\4\5\5\6"
-                                         "\3\4\4\5\4\5\5\6\4\5\5\6\5\6\6\7"
-                                         "\2\3\3\4\3\4\4\5\3\4\4\5\4\5\5\6"
-                                         "\3\4\4\5\4\5\5\6\4\5\5\6\5\6\6\7"
-                                         "\3\4\4\5\4\5\5\6\4\5\5\6\5\6\6\7"
-                                         "\4\5\5\6\5\6\6\7\5\6\6\7\6\7\7\x8";
-        const unsigned char* _Ptr       = &reinterpret_cast<const unsigned char&>(_Array);
-        const unsigned char* const _End = _Ptr + sizeof(_Array);
-        size_t _Val                     = 0;
-        for (; _Ptr != _End; ++_Ptr) {
-            _Val += _Bitsperbyte[*_Ptr];
+#if defined(_M_IX86) || defined(_M_X64)

```

Comment:
Why does this contain the extensive logic needed to directly call the intrinsics, instead of calling `_Popcount()`?

https://github.com/microsoft/STL/blob/f75c7f596c7b491168fefb5eff5164ab7ae36867/stl/inc/limits#L1136-L1147

If the rationale is to avoid repeated tests of `_Definitely_have_popcnt`, I'd recommend profiling the difference. I expect that the test has near-zero cost, as it doesn't invoke any separately compiled functions, and will be perfectly predictable after the first time.

If it has an observable cost, I think it could be better to explore enhancing the `_Popcount()` machinery so we can lift out the check without duplicating the rest of the logic.

Reply:
```C++
// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception

#include <bit>
#include <chrono>
#include <iostream>
#include <random>

const auto length = 10'0000;
const auto repeat = 100;

volatile int isa_available = 6;

template<std::size_t Words>
struct bitset {

    void populate() {
        std::mt19937 mt(43223);
        std::uniform_int_distribution<Ty> dis(std::numeric_limits<Ty>::min(), std::numeric_limits<Ty>::max());
        for (std::size_t Wpos = 0; Wpos <= Words; ++Wpos) {
            Array[Wpos] = dis(mt);
        }
    }

    using Ty = std::size_t;

    size_t count1() const noexcept { // count number of set bits
        Ty Val = 0;
        for (std::size_t Wpos = 0; Wpos <= Words; ++Wpos) {
            Val += std::_Popcount(Val);
        }
        return Val;
    }

    size_t count2() const noexcept { // count number of set bits
        const bool Definitely_have_popcnt = isa_available >= 5;
        if (Definitely_have_popcnt) {
            Ty Val = 0;
            for (std::size_t Wpos = 0; Wpos <= Words; ++Wpos) {
                if constexpr (sizeof(Ty) == 4) {
                    Val += __popcnt(Array[Wpos]);
                }
                else if constexpr (sizeof(Ty) == 8) {
#ifdef _M_IX86
                    _Val += (__popcnt(_Array[_Wpos] >> 32) + __popcnt(static_cast<unsigned int>(_Array[_Wpos])));
#else // ^^^ _M_IX86 / !_M_IX86 vvv
                    Val += __popcnt64(Array[Wpos]);
#endif // _M_IX86
                }
            }
            return Val;
        }

        Ty Val = 0;
        for (std::size_t Wpos = 0; Wpos <= Words; ++Wpos) {
            Val += std::_Popcount_fallback(Array[Wpos]);

        }
        return Val;
    }
    Ty Array[Words+1];
};

volatile std::size_t result;

int main()
{
    const int N = 200000;
    bitset<10'000> bs;
    bs.populate();

    std::chrono::steady_clock::duration d1{};
    std::chrono::steady_clock::duration d2{};
    std::chrono::steady_clock::time_point t;

    t = std::chrono::steady_clock::now();
    for (int i = 0; i < N; ++i) {
        result = bs.count1();
    }
    d1 = std::chrono::steady_clock::now() - t;

    t = std::chrono::steady_clock::now();
    for (int i = 0; i < N; ++i) {
        result = bs.count2();
    }
    d2 = std::chrono::steady_clock::now() - t;

    std::cout << std::chrono::duration_cast<std::chrono::duration<double>>(d1) << " vs " <<
        std::chrono::duration_cast<std::chrono::duration<double>>(d2) << "\n";
    return 0;
}
```

Reply:
2.54269s vs 1.52024s

Reply:
So it makes sense to move dispatch out of loop.
I prefer #2079 and #2127 to land then before doing changes in `std::_Popcount`
Moving to draft for now

Reply:
Found a way to move the check out of loop with minimal `_Popcount` changes. See it working https://godbolt.org/z/EejPGYhr7